File: December - 08/Cafeteria Queue.py
students_line = input().strip()
sandwiches_line = input().strip()
students_str = students_line.split('=',1)[1].strip()
sandwiches_str = sandwiches_line.split('=',1)[1].strip()
students_str = students_str[1:-1]
sandwiches_str = sandwiches_str[1:-1]
students = list(map(int, students_str.split(',')))
sandwiches = list(map(int, sandwiches_str.split(',')))
count0 = students.count(0)
count1 = students.count(1)
for s in sandwiches:
    if s == 0:
        if count0 == 0:
            break
        count0 -= 1
    else:
        if count1 == 0:
            break
        count1 -= 1
print(count0 + count1)

File: December - 08/CafeteriaQueue.java
package Dec08;
import java.util.*;
public class CafeteriaQueue {
    public static void main(String args[]) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int students[] = new int[n];
        int sandwich[] = new int[n];
        for (int i = 0; i < n; i++) {
            students[i] = s.nextInt();
        }
        for(int i=0;i<n;i++){
            sandwich[i]=s.nextInt();
        }
        int count0=0;
        int count1=0;
        for(int i:students){
            if(i==0)
                count0++;
            else count1++;
        }
        for(int i=0;i<n;i++){
            if(sandwich[i]==0){
                if(count0==0)
                    break;
                count0--;
            }
            else{
                if(count1==0)
                    break;
                count1--;
            }
        }
        System.out.println(count0+count1);
    }
}
